matrix/lambdas/api/v1/core.py

```diff
@@ -41,21 +42,38 @@ def post_matrix(body: dict):
                             "Visit https://matrix.dev.data.humancellatlas.org for more information."},
                 requests.codes.request_entity_too_large)
 
-    request_id = str(uuid.uuid4())
-    RequestTracker(request_id).initialize_request(format_, fields, feature)
-
-    driver_payload = {
-        'request_id': request_id,
-        'filter': body["filter"],
-        'fields': fields,
-        'feature': feature
-    }
-    lambda_handler.invoke(LambdaName.DRIVER_V1, driver_payload)
-
-    return ({'request_id': request_id,
+    if query_constructor.has_genus_species_term(body["filter"]):
+        # If the user has mentioned something about species, then maybe
+        # they're looking for non-human data. So we'll run queries for all
+        # the species that we know about.
+        genera_species = list(constants.GenusSpecies)
+    else:
+        # Otherwise, default to human-only
+        genera_species = [constants.GenusSpecies.HUMAN]
+
+    human_request_id = ""
+    non_human_request_ids = {}
+    for genus_species in genera_species:
+        request_id = str(uuid.uuid4())
+        RequestTracker(request_id).initialize_request(format_, fields, feature, genus_species)
+
+        driver_payload = {
+            'request_id': request_id,
+            'filter': body["filter"],
+            'fields': fields,
+            'feature': feature,
+            'genus_species': genus_species.value
+        }
+        lambda_handler.invoke(LambdaName.DRIVER_V1, driver_payload)
```

Comment:
I wonder how much this affects the POST response time if there are multiple species. Something to keep in mind

Reply:
Oh yeah good point. Think it's worth using `ThreadPoolExecutor` or something?